### DISCUSSION TITLE
How to assign event handler to web component?

### DISCUSSION THREAD

**Discussion by Kwooda** [storybookjs/storybook#23344] [Help]
I have a Web Component that has an event called clickEvent. In Storybook 7, I want to simply respond to this event to display an alert, just like I normally would in JavaScript.
I have the following:
```
export const EventHandler = {
	render: () => `	<kds-checkbox onClickEvent="alert('clickEvent')"></kds-checkbox>`,
	name: 'Event Handler',
};
```
When I try to open the story, I get an error:
```
TypeError: Failed to execute 'addEventListener' on 'EventTarget': parameter 2 is not of type 'Object'.
    at Object.ael (p-e63570aa.js:2:190)
```
Interestingly, if I change onClickEvent to onClick, I don't get the error, but I also don't get the right event.
What do I need to do to get Storybook to encode this properly? This is using the @storybook/web-components framework.

**Comment by kenisteward** [comment]
@Kwooda  I know this is really old but this is my suggestion as I've found works best fo rhow we set things up for our we bocmponents.
```
  render: (args, options) => {
    const el = document.createElement('my-web-component');
    el.addEventListener('clickEvent', () => {alert('Hello Alert')});
    return html`${el}`;
  }
```

I've also written an generic function of the preview for the render that converts all onEventName custom-element.json events to actual actions instead of having to tie events directly.  because of this we can test actions using the actions addon or essentials addon instead.

I would also stray on the side of just using the default click event instead of adding a non standard click event.  that way you can jsut use browser support for click natively and prevent default actions with e.stopImmediatePropagation()
